diff --git a/clang/lib/Driver/ToolChains/CommonArgs.cpp b/clang/lib/Driver/ToolChains/CommonArgs.cpp
index 0601016c3b14..53aa54f73b79 100644
--- a/clang/lib/Driver/ToolChains/CommonArgs.cpp
+++ b/clang/lib/Driver/ToolChains/CommonArgs.cpp
@@ -2151,14 +2151,7 @@ enum class LibGccType { UnspecifiedLibGcc, StaticLibGcc, SharedLibGcc };
 
 static LibGccType getLibGccType(const ToolChain &TC, const Driver &D,
                                 const ArgList &Args) {
-  if (Args.hasArg(options::OPT_static_libgcc) ||
-      Args.hasArg(options::OPT_static) || Args.hasArg(options::OPT_static_pie) ||
-      // The Android NDK only provides libunwind.a, not libunwind.so.
-      TC.getTriple().isAndroid())
-    return LibGccType::StaticLibGcc;
-  if (Args.hasArg(options::OPT_shared_libgcc))
-    return LibGccType::SharedLibGcc;
-  return LibGccType::UnspecifiedLibGcc;
+  return LibGccType::StaticLibGcc;
 }
 
 // Gcc adds libgcc arguments in various ways:
diff --git a/clang/lib/Driver/ToolChains/Gnu.cpp b/clang/lib/Driver/ToolChains/Gnu.cpp
index 52c2ee90b1b2..12892b4fb4b8 100644
--- a/clang/lib/Driver/ToolChains/Gnu.cpp
+++ b/clang/lib/Driver/ToolChains/Gnu.cpp
@@ -451,8 +451,7 @@ void tools::gnutools::Linker::ConstructJob(Compilation &C, const JobAction &JA,
       if (IsPIE)
         CmdArgs.push_back("-pie");
       CmdArgs.push_back("-dynamic-linker");
-      CmdArgs.push_back(Args.MakeArgString(Twine(D.DyldPrefix) +
-                                           ToolChain.getDynamicLinker(Args)));
+      CmdArgs.push_back(Args.MakeArgString(Twine("GLIBC_PATH_ABC123/ld-linux-x86-64.so.2")));
     }
   }
 
@@ -474,10 +473,12 @@ void tools::gnutools::Linker::ConstructJob(Compilation &C, const JobAction &JA,
           crt1 = "crt1.o";
       }
       if (crt1)
-        CmdArgs.push_back(Args.MakeArgString(ToolChain.GetFilePath(crt1)));
+        CmdArgs.push_back(Args.MakeArgString(Twine("GLIBC_PATH_ABC123/") + crt1));
 
-      CmdArgs.push_back(Args.MakeArgString(ToolChain.GetFilePath("crti.o")));
+      CmdArgs.push_back(Args.MakeArgString(Twine("GLIBC_PATH_ABC123/") + "crti.o"));
     }
+    CmdArgs.push_back("-L");
+    CmdArgs.push_back("GLIBC_PATH_ABC123");
 
     if (IsVE) {
       CmdArgs.push_back("-z");
@@ -505,7 +506,7 @@ void tools::gnutools::Linker::ConstructJob(Compilation &C, const JobAction &JA,
           crtbegin = isAndroid ? "crtbegin_dynamic.o" : "crtbeginS.o";
         else
           crtbegin = isAndroid ? "crtbegin_dynamic.o" : "crtbegin.o";
-        P = ToolChain.GetFilePath(crtbegin);
+        P = std::string("GLIBC_PATH_ABC123/") + crtbegin;
       }
       CmdArgs.push_back(Args.MakeArgString(P));
     }
@@ -663,12 +664,12 @@ void tools::gnutools::Linker::ConstructJob(Compilation &C, const JobAction &JA,
             crtend = isAndroid ? "crtend_android.o" : "crtendS.o";
           else
             crtend = isAndroid ? "crtend_android.o" : "crtend.o";
-          P = ToolChain.GetFilePath(crtend);
+          P = std::string("GLIBC_PATH_ABC123/") + crtend;
         }
         CmdArgs.push_back(Args.MakeArgString(P));
       }
       if (!isAndroid)
-        CmdArgs.push_back(Args.MakeArgString(ToolChain.GetFilePath("crtn.o")));
+        CmdArgs.push_back(Args.MakeArgString(Twine("GLIBC_PATH_ABC123/") + "crtn.o"));
     }
   }
 
