--- a/pkg/builds/cuda_nvcc/include/crt/host_defines.h
+++ b/pkg/builds/cuda_nvcc/include/crt/host_defines.h
@@ -60,14 +60,6 @@
 #if !defined(__HOST_DEFINES_H__)
 #define __HOST_DEFINES_H__
 
-#if defined(__CUDACC__) && !defined(__CUDACC_RTC__) && !defined(__CUDADEVRT_INTERNAL__) && !defined(_ALLOW_UNSUPPORTED_LIBCPP)
-#include <ctype.h>
-#if ((defined(_MSC_VER ) && (defined(_M_X64) || defined(_M_AMD64))) ||\
-     (defined(__x86_64__) || defined(__amd64__))) && defined(_LIBCPP_VERSION) && !(defined(__HORIZON__) || defined(__ANDROID__) || defined(__QNX__))
-#error "libc++ is not supported on x86 system"
-#endif
-#endif
-
 /* CUDA JIT mode (__CUDACC_RTC__) also uses GNU style attributes */
 #if defined(__GNUC__) || (defined(__PGIC__) && defined(__linux__)) || defined(__CUDA_LIBDEVICE__) || defined(__CUDACC_RTC__)
 
